test/api/v3/integration/groups/GET-groups.test.js

```diff
@@ -0,0 +1,97 @@
+import {
+  generateUser,
+  resetHabiticaDB,
+  generateGroup,
+} from '../../../../helpers/api-v3-integration.helper';
+
+describe('GET /groups', () => {
+  let user;
+  const NUMBER_OF_PUBLIC_GUILDS = 3;
+  const NUMBER_OF_USERS_PRIVATE_GUILDS = 1;
+  const NUMBER_OF_GUILDS_USER_CAN_VIEW = 6;
+
+  before(async () => {
+    await resetHabiticaDB();
```

Comment:
It is necessary. Any other test that creates a public guild before this test runs will have that included in the public guild query. Resetting the db to an empty (plus tavern) state ensures that only the newly created public guilds will be counted.